src/components/clusterConfiguration/discoveryTroubleshootingModal.tsx

```diff
@@ -0,0 +1,134 @@
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+  const closeModal = () => setIsModalOpen(false);
+  const handleClick = onClick || (() => setIsModalOpen(true));
+  return (
+    <>
+      <ButtonComponent variant={ButtonVariant.link} {...props} onClick={handleClick} isInline>
+        {children}
+      </ButtonComponent>
+      <DiscoveryTroubleshootingModal isOpen={isModalOpen} onClose={closeModal} />
+    </>
+  );
+};
+
+type DiscoveryTroubleshootingModalProps = {
+  onClose: () => void;
+  isOpen: boolean;
+};
+
+export const DiscoveryTroubleshootingModal: React.FC<DiscoveryTroubleshootingModalProps> = ({
+  onClose,
+  isOpen,
+}) => {
+  return (
+    <Modal
+      title="Host Discovery Troubleshooting"
+      isOpen={isOpen}
+      actions={[
+        <Button key="close" variant={ButtonVariant.primary} onClick={onClose}>
+          Close
+        </Button>,
+      ]}
+      onClose={onClose}
+      variant={ModalVariant.large}
+    >
+      <TextContent>
+        <Text component={TextVariants.p}>
+          Booting up an ISO might take a few minutes, and might depend heavily on your hardware and
+          network configuration, however, if after a few minutes your machine does not show up in
+          the UI, it's probably best to try and login to it via SSH, for that you would need to know
+          the machine ip-address, or use a console of some sort.
+        </Text>
+        <Text component={TextVariants.h2}>SSH into your machine</Text>
+        <PrismCode code={`ssh core@<machine-ip>`} />
+        <Text component={TextVariants.p}>
+          If SSH key was provided when generating the ISO it is not required to provide a password.
+        </Text>
+        <Text component={TextVariants.h2}>
+          Verify that the discovery agent is running correctly
+        </Text>
+        <Text component={TextVariants.p}>Execute</Text>
+        <PrismCode code={`ps -ef | grep agent`} />
+        <Text component={TextVariants.p}>
+          and see whether the agent runs with correct parameters. The result should look similar to
+          this:
+        </Text>
+        <PrismCode
+          code={`root        1342       1  0 09:56 ?        00:00:00 /usr/local/bin/agent --host 192.168.39.162 --port 30956 --cluster-id a8142e14-8bfe-46bd-bfbd-4bd1a126892b core        2063    2003  0 10:41 pts/0    00:00:00 grep --color=auto agent`}
+        />
+        <Text component={TextVariants.p}>
+          See that logs are indicating that agent was run successfully:
+        </Text>
+        <PrismCode
+          code={`[systemUser@vm-11-47 ~]$ sudo journalctl -u agent
+-- Logs begin at Mon 2020-05-25 14:02:10 UTC, end at Mon 2020-05-25 19:50:52 UTC. --
+May 25 14:02:25 localhost systemd[1]: Starting agent.service...
+May 25 14:02:31 vm-11-47.lab.eng.tlv2.redhat.com docker[957]: Unable to find image 'quay.io/oamizur/agent:latest' locally
+May 25 14:02:33 vm-11-47.lab.eng.tlv2.redhat.com docker[957]: latest: Pulling from oamizur/agent
```

Comment:
i might remove hostnames, references to quay images (as it will change)

Reply:
done